polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -2143,8 +2141,34 @@ MLIRASTConsumer::getOrCreateGlobal(const ValueDecl &VD, std::string Prefix,
     break;
   }
 
-  // Initialize the global variable.
-  if (const Expr *Init = Var->getInit()) {
+  // Initialize the global.
+  const VarDecl *InitDecl;
+  const Expr *InitExpr = Var->getAnyInitializer(InitDecl);
```

Comment:
As `InitDecl` is not used, you can probably use:
```
const Expr *getAnyInitializer() const {
    const VarDecl *D;
    return getAnyInitializer(D);
  }
  ```
```suggestion
  const Expr *InitExpr = Var->getAnyInitializer();
```